packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2297,6 +2325,37 @@ Hardhat Network's forking functionality only works with blocks from at least spu
 
     return { maxFeePerGas, maxPriorityFeePerGas };
   }
+
+  /** search this._hardforkActivationBlocks for the highest block number that
+   * isn't higher than blockNum, and then return that found block number's
+   * associated hardfork name. */
+  private _selectHardforkFromActivations(
+    blockNum: BN
+  ): typeof HARDHAT_NETWORK_SUPPORTED_HARDFORKS[number] {
+    let highestFound: {
```

Comment:
If we're not running a forked node, we should immediately return the local hardfork, the one that was used in the constructor parameter for the node.

Reply:
Also need to just use the constructor-given hardfork in the case where the block number to execute against is higher than the block number that was forked from.